core/utils/ethabi.go

```diff
@@ -14,8 +15,26 @@ import (
 )
 
 const (
-	// FormatBytes encodes the output as bytes
+	// FormatRawHexWithFuncSelectorAndDataPrefix is the default format behavior.
+	// Its output is prefixed with the ethTx function selector, followed by the
+	// data prefix
+	FormatRawHexWithFuncSelectorAndDataPrefix = ""
+	// FormatBytes encodes the output as bytes. I.e., the string input will be
+	// cast to bytes, and passed to the on-chain contract method as a solidity
+	// bytes array argument. (No conversion from hex is done; the string input
+	// must be the raw bytes!)
 	FormatBytes = "bytes"
+	// FormatPreformatted encodes the output, assumed to be hex, as bytes and
+	// passes them as arguments. Caller is responsible for all formatting for the
+	// EVM. Input must be 0x-prefixed
+	FormatPreformattedHexArguments = "preformattedHexArguments"
+	// FormatRawHex does no formatting at all. Caller is responsible for
+	// formatting the function selector and offset, in addition to any arguments
+	// to be passed with the transaction. Input must be 0x-prefixed
+	//
+	// Note that this option isn't adressed in EVMTranscodeJSONWithFormat, because
+	// eth_tx.go's getTxData short-circuits, when it encounters this.
+	FormatRawHex = "rawHex"
```

Comment:
This should have a test case.

Reply:
[Is this inadequate?](https://github.com/smartcontractkit/chainlink/pull/2291/files#diff-0d44f8ae519e9f175ac946337ffb211bR94)